luaui/Widgets/gui_nanotrans_range.lua

```diff
@@ -0,0 +1,97 @@
+function widget:GetInfo()
+    return {
+       name         = "Nano range on transport",
+       desc         = "Draw a circle around a transport carrying a nano when its about to unload",
+       author       = "Cheva",
+       date         = "December 2024",
+       license      = "GNU GPL, v2 or later",
+       layer        = 0,
+       enabled      = true
+    }
+end
+
+--------------------------------------------------------------------------------
+--vars
+--------------------------------------------------------------------------------
+local circleDivs = 96
+local range
+local isNanoTC = {}
+local turretRange = {}
+local transportWithNano = {}
+
+--------------------------------------------------------------------------------
+--speedups
+--------------------------------------------------------------------------------
+local CMD_UNLOAD_UNITS = CMD.UNLOAD_UNITS
+local spGetActiveCmd = Spring.GetActiveCommand
+local GetSelectedUnitsSorted = Spring.GetSelectedUnitsSorted
+local glColor = gl.Color
+local glLineWidth = gl.LineWidth
+local glDrawGroundCircle = gl.DrawGroundCircle
+
+for udid, ud in pairs(UnitDefs) do
+    if ud.isStaticBuilder and not ud.isFactory then
+        isNanoTC[udid] = true
+        turretRange[udid] = ud.buildDistance
+    end
+end
+
+--------------------------------------------------------------------------------
+-- utility functions
+--------------------------------------------------------------------------------
+function MinValue(table)
+    local min = table[1]
+    for i = 2, #table do
+        if table[i] < min then
+            min = table[i]
+        end
+    end
+    return min
+end
```

Comment:
Try to use an existing common table function inside https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/common/tablefunctions.lua. If no suitable function exists, adds the new functionality there.

Reply:
I remove the function and found a new way to get the minimum